airbyte-cdk/bulk/toolkits/load-object-storage/src/testFixtures/kotlin/io/airbyte/cdk/load/ObjectStorageDataDumper.kt

```diff
@@ -50,11 +51,19 @@ class ObjectStorageDataDumper(
     private val parquetMapperPipeline = ParquetMapperPipelineFactory().create(stream)
 
     fun dump(): List<OutputRecord> {
-        val prefix = pathFactory.getFinalDirectory(stream).toString()
+        // Note: this is implicitly a test of the `streamConstant` final directory
+        // and the path matcher, so a failure here might imply a bug in the metadata-based
+        // destination state loader, which lists by `prefix` and filters against the matcher.
+        val prefix =
+            pathFactory.getFinalDirectory(stream, streamConstant = true).toString().takeWhile {
+                it != '$'
+            }
```

Comment:
Nit: should we share this code since it's coupled?

Reply:
yeah, it should really go into the path factory, and be like "longestStreamConstantPrefix" or something